### DISCUSSION TITLE
Metals not detecting scala.App in VS Code

### DISCUSSION THREAD

**Discussion by nestorsokil** [scalameta/metals#6774] [General]
Hey, I admit I am a little new to Scala, but I have an existing project, and its main class is mixing the trait `scala.App`. The problem is Metals is not showing the "Run" popup, i.e., the green arrow. It does detect the project structure though just no option to run and debug.

**Comment by tgodzik** [comment]
Thanks for the question!

If the code compiles, there should be `run | debug` code lenses show up above the main class. If it's not happening it might be worth checking the doctor (via show doctor command or via a shortcut in the metals tab).

Also if there is nothing there it might be worth posting the project for reproduction or at least the logs available in `.metals/metals.log`

**Reply by nestorsokil** [reply]
The code does compile with maven and doctor shows all modules as compiled ✅ but I'm not sure if metals/bloop is able to handle it as it is a mixed Java/Scala project.

**Reply by tgodzik** [reply]
It should be good then, nothing in those logs I mentioned?
